packages/react/src/Button/Button.tsx

```diff
@@ -6,26 +6,42 @@
 import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ButtonHTMLAttributes, ForwardedRef, PropsWithChildren } from 'react'
+import { Icon } from '../Icon'
+
+type ButtonIcon = {
+  icon?: Function
+  iconStart?: never
+  iconOnly?: boolean
+}
+
+type ButtonIconStart = {
+  icon?: never
+  iconStart?: Function
+  iconOnly?: boolean
+}
 
 export type ButtonProps = {
   /** The level of prominence. Use a primary button only once per page or section. */
   variant?: 'primary' | 'secondary' | 'tertiary'
-} & PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>>
+} & (ButtonIcon | ButtonIconStart) &
+  PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>>
 
 export const Button = forwardRef(
   (
-    { children, className, type, disabled, variant = 'primary', ...restProps }: ButtonProps,
+    { children, className, type, disabled, variant = 'primary', icon, iconStart, iconOnly, ...restProps }: ButtonProps,
```

Comment:
Please sort the props

Reply:
Sorted, this is something I forget when the props are formatted (stylelint) on one line.